includes/compat/subscriptions/class-wc-payment-gateway-wcpay-subscriptions-compat.php

```diff
@@ -79,14 +80,18 @@ private function is_changing_payment_method_for_subscription() {
 	/**
 	 * Process the payment for a given order.
 	 *
-	 * @param int  $order_id Order ID to process the payment for.
-	 * @param bool $is_recurring_payment Whether this is a one-off payment (false) or it's the first installment of a recurring payment (true).
+	 * @param int    $order_id Order ID to process the payment for.
+	 * @param bool   $force_save_payment_method Whether to save the payment method upon successful payment.
+	 * @param string $payment_type              The type of the payment that's being made.
 	 *
 	 * @return array|null An array with result of payment and redirect URL, or nothing.
 	 */
-	public function process_payment( $order_id, $is_recurring_payment = false ) {
-		$force_save_payment_method = wcs_order_contains_subscription( $order_id ) || $this->is_changing_payment_method_for_subscription();
-		return parent::process_payment( $order_id, $force_save_payment_method );
+	public function process_payment( $order_id, $force_save_payment_method = false, $payment_type = null ) {
+		if ( wcs_order_contains_subscription( $order_id ) || $this->is_changing_payment_method_for_subscription() ) {
+			return parent::process_payment( $order_id, true, Payment_Type::RECURRING() );
+		} else {
+			return parent::process_payment( $order_id, $force_save_payment_method, $payment_type );
+		}
```

Comment:
can we revert the checks to make it clear where we fully fallback to the parent and where subs specific code is implemented?

```suggestion
		if ( ! wcs_order_contains_subscription( $order_id ) && ! $this->is_changing_payment_method_for_subscription() ) {
			return parent::process_payment( $order_id, $force_save_payment_method, $payment_type );
		}
		
		// Subs specific behavior starts here.
		return parent::process_payment( $order_id, true, Payment_Type::RECURRING() );
```